www/include/monitoring/status/ServicesHostGroups/xml/serviceGridByHGXML.php

```diff
@@ -53,18 +53,26 @@
 // Set Default Poller
 $obj->getDefaultFilters();
 
-// Check Arguments From GET tab
+/*
+ *  Check Arguments from GET and session
+ */
+// integer values from $_GET
+$p = filter_input(INPUT_GET, 'p', FILTER_VALIDATE_INT, array('options' => array('default' => 2)));
+$num = filter_input(INPUT_GET, 'num', FILTER_VALIDATE_INT, array('options' => array('default' => 0)));
+$limit = filter_input(INPUT_GET, 'limit', FILTER_VALIDATE_INT, array('options' => array('default' => 20)));
+
+// use whitelist values instead of the $_GET value
+$order = isset($_GET['order']) && $_GET['order'] === "DESC" ? "DESC" : "ASC";
```

Comment:
you could use filter_input with FILTER_VALIDATE_REGEXP flag

Reply:
wouldn't a regexp consume more ressource than a single whitelist string check ?
as there's only two possibilities, and by default ASC should be set

Reply:
a simple regex does not consume lot of resource
It's not for this kind of stuff which we need to look at performance issues

Reply:
done